synapse/models/dns.py

```diff
@@ -1,75 +1,111 @@
 import synapse.lib.module as s_module
 
-class DnsMod(s_module.CoreModule):
-
-    @staticmethod
-    def getBaseModels():
+class DnsModule(s_module.CoreModule):
+    def getModelDefs(self):
         modl = {
+            'types': (
+                ('inet:dns:a', ('comp', {'fields': (('fqdn', 'inet:fqdn'), ('ipv4', 'inet:ipv4'))}), {
+                    'doc': 'The result of a DNS A record lookup.'
+                }),
+                ('inet:dns:aaaa', ('comp', {'fields': (('fqdn', 'inet:fqdn'), ('ipv6', 'inet:ipv6'))}), {
+                    'doc': 'The result of a DNS AAAA record lookup.'
+                }),
+                ('inet:dns:rev', ('comp', {'fields': (('ipv4', 'inet:ipv4'), ('fqdn', 'inet:fqdn'))}), {
+                    'doc': 'The transformed result of a DNS PTR record lookup.'
+                }),
+                ('inet:dns:rev6', ('comp', {'fields': (('ipv6', 'inet:ipv6'), ('fqdn', 'inet:fqdn'))}), {
+                    'doc': 'The transformed result of a DNS PTR record for an IPv6 address.'
+                }),
+                ('inet:dns:ns', ('comp', {'fields': (('zone', 'inet:fqdn'), ('ns', 'inet:fqdn'))}), {
+                    'doc': 'The result of a DNS NS record lookup.'
+                }),
+                ('inet:dns:cname', ('comp', {'fields': (('fqdn', 'inet:fqdn'), ('cname', 'inet:fqdn'))}), {
+                    'doc': 'The result of a DNS CNAME record lookup.'
+                }),
+                ('inet:dns:mx', ('comp', {'fields': (('fqdn', 'inet:fqdn'), ('mx', 'inet:fqdn'))}), {
+                    'doc': 'The result of a DNS MX record lookup.'
+                }),
+                ('inet:dns:soa', ('guid', {}), {
+                    'doc': 'The result of a DNS SOA record lookup.'
+                }),
+                ('inet:dns:txt', ('comp', {'fields': (('fqdn', 'inet:fqdn'), ('txt', 'str'))}), {
+                    'doc': 'The result of a DNS MX record lookup.'
+                }),
+            ),
+            'forms': (
+                ('inet:dns:a', {}, (
+                    ('fqdn', ('inet:fqdn', {}), {'ro': 1,
+                        'doc': 'The domain queried for its DNS A record.'}),
+                    ('ipv4', ('inet:ipv4', {}), {'ro': 1,
+                        'doc': 'The IPv4 address returned in the A record.'}),
+                    # FIXME - add Seen:min / seen:max
```

Comment:
`seen:min`/`seen:max` are currently available

Reply:
punting on these for now